test/core/test/jest-expect.test.ts

```diff
@@ -1849,3 +1849,35 @@ it('diff', () => {
   snapshotError(() => expect({ hello: 'world' }).toBeUndefined())
   snapshotError(() => expect({ hello: 'world' }).toBeNull())
 })
+
+describe('undefined property handling', () => {
+  it('toEqual ignores undefined properties', () => {
+    const obj1 = {
+      foo: 'bar',
+      qux: 'qux',
+    }
+
+    const obj2 = {
+      foo: 'bar',
+      extra: undefined,
+      qux: 'qux',
+    }
+
+    expect(obj1).toEqual(obj2)
```

Comment:
Can you add tests for `toEqual` error diff? I think that's the issue raised by https://github.com/vitest-dev/vitest/issues/7083 

```js
snapshotError(() =>
  expect({
    x: 'foo',
    y: undefined,
  }).toEqual({
    x: 'bar',
  })
);

snapshotError(() =>
  expect({
    x: 'foo',
  }).toEqual({
    x: 'bar',
    y: undefined,
  })
);
```